loadbalancer/etcd/client.go

```diff
@@ -25,27 +30,69 @@ type client struct {
 // The parameter machines needs to be a full URL with schemas.
 // e.g. "http://localhost:4001" will work, but "localhost:4001" will not.
 func NewClient(machines []string, cert, key, caCert string) (Client, error) {
-	var c *etcd.Client
-	var err error
+	var c etcd.KeysAPI
 
 	if cert != "" && key != "" {
-		c, err = etcd.NewTLSClient(machines, cert, key, caCert)
+
+		tlsCert, err := tls.LoadX509KeyPair(cert, key)
+		if err != nil {
+			return nil, err
+		}
+
+		caCertCt, err2 := ioutil.ReadFile(caCert)
+		if err2 != nil {
+			return nil, err2
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCertCt)
+
+		tlsConfig := &tls.Config{
+			Certificates: []tls.Certificate{tlsCert},
+			RootCAs:      caCertPool,
+		}
+
+		transport := &http.Transport{
+			TLSClientConfig: tlsConfig,
+			Dial: func(network, addr string) (net.Conn, error) {
+				dialer := net.Dialer{
+					Timeout:   time.Second,
+					KeepAlive: time.Second,
```

Comment:
Where do these values come from? Please make them functional options. This will probably require introducing an options struct.